Dockerfile

```diff
@@ -1,58 +1,101 @@
-FROM ubuntu:18.04
+# should match the GHC version of the stack.yaml resolver
+FROM haskell:8.10.4 AS ihaskell_base
```

Comment:
This still needs to be manually updated, right? Or am I missing something?

Reply:
oh right sure. i can make that automatic too; it would just make the `docker build` command a bit more complicated. lemme update the PR

Reply:
How are you currently continuously-deploying the docker image? Is it via CI or Docker Hub's build feature?

I usually solve this problem by adding an `ARG` and writing a wrapper script that reads `stack.yaml` and sets the `--build-arg` automatically, but I don't think that would work if you deploy with Docker Hub. If you use Docker Hub, what we can do instead is keep this hardcoded, then add a test in CI to check that it matches `stack.yaml`. So at least there'd be an alert

Reply:
I'm using CI to update the Docker image every time `master` is updated: [docker.yml](https://github.com/gibiansky/IHaskell/blob/6e69c210c5127d74870a8d3645fc492a7f064eba/.github/workflows/docker.yml)